presto-cluster-ttl-providers/src/main/java/com/facebook/presto/clusterTTLProviders/percentile/PercentileBasedClusterTTLProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.clusterTTLProviders.percentile;
+
+import com.facebook.presto.spi.ttl.ClusterTTLProvider;
+import com.facebook.presto.spi.ttl.ConfidenceBasedTTLInfo;
+import com.facebook.presto.spi.ttl.NodeTTL;
+import com.google.inject.Inject;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class PercentileBasedClusterTTLProvider
+        implements ClusterTTLProvider
+{
+    private final Comparator<ConfidenceBasedTTLInfo> ttlComparator = Comparator.comparing(ConfidenceBasedTTLInfo::getExpiryInstant);
```

Comment:
As it stands, the objective of the `PercentileBasedClusterTTLProvider` is to provide a TTL for the cluster based purely on the `expiryInstant`s of the workers in the cluster. In this initial iteration, I want to keep the comparison simple. In the future, we can include the `confidencePercentage` in the comparison and it would probably reduce the `expiryInstant` of a worker for any value < 100% (this would also be custom logic based on what confidence means for that particular TTL fetcher).